# **To retain the maximum amount of training data for model training**

In [1]:
import pandas as pd
import numpy as np

In [3]:
# Read the original training dataset
df_train_set = pd.read_csv('../../../../../data/drug full screening data NEW edited.csv')
df_train_set

DRUG NAME  MDA7+D1+1  MDA7+D1+2  MDA7+D1+3  MDA7+D2+1  \
0             Angiotensin II   0.467198   0.492122   0.500198   0.651233   
1              Levetiracetam   0.424844   0.503705   0.433523   0.517083   
2                 Daptomycin   0.462072   0.509482   0.424083   0.451883   
3               Lansoprazole   0.464806   0.494925   0.502533   0.491361   
4                  Adapalene   0.407362   0.425062   0.490854   0.489590   
...                      ...        ...        ...        ...        ...   
1966     Chondroitin sulfate   0.750820   0.741770   0.826674   0.835227   
1967                 Aucubin   0.714064   0.849337   0.563837   0.869225   
1968           Cepharanthine   0.592418   0.708048   0.883598   0.811905   
1969     4-Aminobutyric acid   0.705559   0.941525   0.717699   0.887134   
1970  Prulifloxacin (Pruvel)   0.844985   0.773646   0.742799   0.802596   

      MDA7+D2+2  MDA7+D2+3      D1+1      D1+2      D1+3  ...  MDA7+D1+9  \
0      0.762780   0.484070  0.878523  0.952131  1.122307  ...        NaN   
1      0.498133   0.485443  0.985621  1.002564  1.297499  ...        NaN   
2      0.495565   0.472871  0.980899  0.906619  0.982075  ...        NaN   
3      0.479975   0.443513  1.363714  0.919337  1.273511  ...        NaN   
4      0.513733   0.395002  0.437537  0.569486  0.435191  ...        NaN   
...         ...        ...       ...       ...       ...  ...        ...   
1966   0.807444   0.802552  0.790976  0.905025  0.865902  ...        NaN   
1967   0.700734   0.823960  0.808023  0.719939  0.689561  ...        NaN   
1968   0.849278   0.827013  1.053463  1.007266  0.957949  ...        NaN   
1969   0.716770   0.617353  0.804578  0.716076  0.845920  ...        NaN   
1970   0.787334   0.757312  0.618461  0.797263  0.793098  ...        NaN   

      MDA7+D2+7  MDA7+D2+8  MDA7+D2+9  D1+7  D1+8  D1+9  D2+7  D2+8  D2+9  
0           NaN        NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN  
1           NaN        NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN  
2           NaN        NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN  
3           NaN        NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN  
4           NaN        NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN  
...         ...        ...        ...   ...   ...   ...   ...   ...   ...  
1966        NaN        NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN  
1967        NaN        NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN  
1968        NaN        NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN  
1969        NaN        NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN  
1970        NaN        NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN  

[1971 rows x 37 columns]

In [6]:
# Read the FDA-approved drug library datset
df_fda_library = pd.read_csv('../../../../../data/L1021-DiscoveryProbe-FDA-approved-drug-Library 1971 COMPOUNDS_cleaned.csv')
df_fda_library

CatalogNumber               Item Name   CAS Number Plate Location  \
0            A1042          Angiotensin II    4474-91-3            A01   
1            A1198           Levetiracetam  102767-28-2            A02   
2            A1206              Daptomycin  103060-53-3            A03   
3            A1229            Lansoprazole  103577-45-3            A04   
4            A1267               Adapalene  106685-40-9            A05   
...            ...                     ...          ...            ...   
1966         N2730     Chondroitin sulfate    9007-28-7            G04   
1967         N2758                 Aucubin     479-98-1            H01   
1968         N2771           Cepharanthine     481-49-2            H02   
1969        P10002     4-Aminobutyric acid      56-12-2            H03   
1970        P10019  Prulifloxacin (Pruvel)  123447-62-1            H04   

     Rack Number     M.w.         Formula  \
0       L1021-01   1046.2    C50H71N13O12   
1       L1021-01   170.21       C8H14N2O2   
2       L1021-01  1620.67   C72H101N17O26   
3       L1021-01   369.36   C16H14F3N3O2S   
4       L1021-01   412.52        C28H28O3   
...          ...      ...             ...   
1966    L1021-23   489.45  (C14H21NO14S)n   
1967    L1021-23   346.33        C15H22O9   
1968    L1021-23   606.71      C37H38N2O6   
1969    L1021-23   103.12         C4H9NO2   
1970    L1021-23   461.46    C21H20FN3O6S   

                                                 SMILES  \
0                                                   NaN   
1                                  CCC(C(=O)N)N1CCCC1=O   
2     CCCCCCCCCC(=O)NC(CC1=CNC2=CC=CC=C21)C(=O)NC(CC...   
3       CC1=C(C=CN=C1CS(=O)C2=NC3=CC=CC=C3N2)OCC(F)(F)F   
4     COC1=C(C=C(C=C1)C2=CC3=C(C=C2)C=C(C=C3)C(=O)O)...   
...                                                 ...   
1966                                                NaN   
1967                                                NaN   
1968  CN1CCC2=CC3=C(C4=C2C1CC5=CC=C(C=C5)OC6=C(C=CC(...   
1969                                       C(CC(=O)O)CN   
1970                                                NaN   

                                  Solubility       Form          Pathway  \
0     ≥100.2mg/mL in H2O, <2.09mg/mL in DMSO        NaN   Cancer Biology   
1                          >8.5mg/mL in DMSO  Free Base   Cancer Biology   
2                         >81.1mg/mL in DMSO  Free Base   Cancer Biology   
3                         >18.5mg/mL in DMSO  Free Base   Cancer Biology   
4                         >10.3mg/mL in DMSO  Free Base   Cancer Biology   
...                                      ...        ...              ...   
1966                                     NaN        NaN  Nature Products   
1967                      ≥34.6mg/mL in DMSO        NaN  Nature Products   
1968                      >21.9mg/mL in DMSO  Free Base  Nature Products   
1969                       ≥10.3mg/mL in H2O        NaN   Other Products   
1970     ≥23.05mg/mL in DMSO with ultrasonic        NaN   Other Products   

                         Target            Indication  \
0                Cardiovascular                   NaN   
1     Other Signal Transduction  Neurological Disease   
2                 DNA Synthesis             Infection   
3                   Proton Pump             Infection   
4                        Others          Inflammation   
...                         ...                   ...   
1966                        NaN                         
1967                        NaN                   NaN   
1968                        NaN     Metabolic Disease   
1969                       APIs                   NaN   
1970                       APIs                   NaN   

                       Information  \
0                              NaN   
1               Antiepileptic drug   
2     Calcium-dependent antibiotic   
3           H+,K+-ATPase inhibitor   
4            RARβ and RARγ agonist   
...                            ...   
19

In [7]:
# Print the columns
df_fda_library.columns

Index(['CatalogNumber', 'Item Name', 'CAS Number', 'Plate Location',
       'Rack Number', 'M.w.', 'Formula', 'SMILES', 'Solubility', 'Form',
       'Pathway', 'Target', 'Indication', 'Information', 'URL'],
      dtype='object')

In [8]:
# Select the important columns
selection = ['CatalogNumber', 'Item Name', 'M.w.', 'Formula', 'SMILES', 'URL']
df_fda_selected = df_fda_library[selection]
df_fda_selected

CatalogNumber               Item Name     M.w.         Formula  \
0            A1042          Angiotensin II   1046.2    C50H71N13O12   
1            A1198           Levetiracetam   170.21       C8H14N2O2   
2            A1206              Daptomycin  1620.67   C72H101N17O26   
3            A1229            Lansoprazole   369.36   C16H14F3N3O2S   
4            A1267               Adapalene   412.52        C28H28O3   
...            ...                     ...      ...             ...   
1966         N2730     Chondroitin sulfate   489.45  (C14H21NO14S)n   
1967         N2758                 Aucubin   346.33        C15H22O9   
1968         N2771           Cepharanthine   606.71      C37H38N2O6   
1969        P10002     4-Aminobutyric acid   103.12         C4H9NO2   
1970        P10019  Prulifloxacin (Pruvel)   461.46    C21H20FN3O6S   

                                                 SMILES  \
0                                                   NaN   
1                                  CCC(C(=O)N)N1CCCC1=O   
2     CCCCCCCCCC(=O)NC(CC1=CNC2=CC=CC=C21)C(=O)NC(CC...   
3       CC1=C(C=CN=C1CS(=O)C2=NC3=CC=CC=C3N2)OCC(F)(F)F   
4     COC1=C(C=C(C=C1)C2=CC3=C(C=C2)C=C(C=C3)C(=O)O)...   
...                                                 ...   
1966                                                NaN   
1967                                                NaN   
1968  CN1CCC2=CC3=C(C4=C2C1CC5=CC=C(C=C5)OC6=C(C=CC(...   
1969                                       C(CC(=O)O)CN   
1970                                                NaN   

                                                  URL  
0      http://www.apexbt.com/search.php?catalog=A1042  
1      http://www.apexbt.com/search.php?catalog=A1198  
2      http://www.apexbt.com/search.php?catalog=A1206  
3      http://www.apexbt.com/search.php?catalog=A1229  
4      http://www.apexbt.com/search.php?catalog=A1267  
...                                               ...  
1966   http://www.apexbt.com/search.php?catalog=N2730  
1967   http://www.apexbt.com/search.php?catalog=N2758  
1968   http://www.apexbt.com/search.php?catalog=N2771  
1969  http://www.apexbt.com/search.php?catalog=P10002  
1970  http://www.apexbt.com/search.php?catalog=P10019  

[1971 rows x 6 columns]

In [9]:
# Count the NA counts in smiles
na_smiles_count = df_fda_selected.SMILES.isna().value_counts()
na_smiles_count

False    1743
True      228
Name: SMILES, dtype: int64

**There are 228 NA counts in SMILES column** - This has to be filled properly

## Select the corresponding **CatalogNumber** for NA smiles to retrieve

In [10]:
# Select the CatalogNumber
num_catalog_list = []
for i in range(len(df_fda_selected)):
    if df_fda_selected.SMILES[i] is np.nan:
        num_catalog_list.append(df_fda_selected.CatalogNumber[i])
print(len(num_catalog_list))
print(num_catalog_list)

228
['A1042', 'A1736', 'A2552', 'A2614', 'A2664', 'A2877', 'A3151', 'A3191', 'A3209', 'A3327', 'A3348', 'A3371', 'A3381', 'A3392', 'A3412', 'A3496', 'A3513', 'A3543', 'A3631', 'A3664', 'A3758', 'A3891', 'A3892', 'A4014', 'A4024', 'A4072', 'A4078', 'A4153', 'A4227', 'A4323', 'A4448', 'A4468', 'A7048', 'A7486', 'A8309', 'A8347', 'A8364', 'A8377', 'A8384', 'A8390', 'A8408', 'A8443', 'A8452', 'A8460', 'A8686', 'A8704', 'B1027', 'B1054', 'B1070', 'B1125', 'B1199', 'B1223', 'B1286', 'B1303', 'B1392', 'B1458', 'B1469', 'B1477', 'B1505', 'B1514', 'B1551', 'B1598', 'B1628', 'B1639', 'B1647', 'B1648', 'B1671', 'B1709', 'B1716', 'B1717', 'B1729', 'B1747', 'B1793', 'B1797', 'B1823', 'B1954', 'B1999', 'B2026', 'B2086', 'B2123', 'B2142', 'B2241', 'B2258', 'B2295', 'B3234', 'B3267', 'B3283', 'B3304', 'B3341', 'B3344', 'B3484', 'B3530', 'B3674', 'B4720', 'B4721', 'B4723', 'B4737', 'B4745', 'B4786', 'B4806', 'B4882', 'B4883', 'B4950', 'B4964', 'B4965', 'B4968', 'B4975', 'B5736', 'B5864', 'B5888', 'B595

In [13]:
# Retrieve the missing SMILES from url given
import requests
from bs4 import BeautifulSoup

smiles_list = []
badi = []

for i in range(len(num_catalog_list)):
    url = 'http://www.apexbt.com/search.php?catalog='
    url = url + str(num_catalog_list[i])
    # print(url)
    try: 
        response = requests.get(url)
        soup = BeautifulSoup(response.text, 'html.parser')
        row = soup.find('td', string='Canonical SMILES').find_next_sibling('td')
        smiles_list.append(row.text)
    except:
        print("Unexpected error occured at " + str(i) + "th iteration, catalog number: " + str(num_catalog_list[i]) + ".")
        badi.append(i)
        smiles_list.append('NA')

print(len(smiles_list))

Unexpected error occured at 19th iteration, catalog number: A3664.
Unexpected error occured at 25th iteration, catalog number: A4072.


In [11]:
print(len(smiles_list), len(badi))

228 22


In [12]:
# Create dataframe with catalog number and smiles
df_smiles = pd.DataFrame({'CatalogNumber': num_catalog_list, 'SMILES': smiles_list})
df_smiles

CatalogNumber                                             SMILES
0           A1042  CCC(C)C(C(=O)NC(CC1=CN=CN1)C(=O)N2CCCC2C(=O)NC...
1           A1736  C1C(COC2=NC(=CN21)[N+](=O)[O-])OCC3=CC=C(C=C3)...
2           A2552  C1=CC(=C(C=C1C2=C(C(=O)C3=C(C=C(C=C3O2)O)O)O)O...
3           A2614  B(C(CC(C)C)NC(=O)C(CC1=CC=CC=C1)NC(=O)C2=NC=CN...
4           A2664  CC(C)(CO)C1=CC2=CC(=C(C=C2N1CC(CO)O)F)NC(=O)C3...
..            ...                                                ...
223         N2591  O[C@@]1(CC)C2=C(COC1=O)C(N(CC3=C4N=C(C=C5)C(C=...
224         N2706                          O[C@H](C(C=C1)=CC=C1O)CNC
225         N2730                                                 NA
226         N2758                                                 NA
227        P10019  CC1N2C3=CC(=C(C=C3C(=O)C(=C2S1)C(=O)O)F)N4CCN(...

[228 rows x 2 columns]

In [13]:
# Add the smiles to the original dataframe
for i in range(len(df_fda_selected)):
    for j in range(len(df_smiles)):
        if df_fda_selected.CatalogNumber[i] == df_smiles.CatalogNumber[j]:
            df_fda_selected.SMILES[i] = df_smiles.SMILES[j]
df_fda_selected

c:\Anaconda3\envs\dleps\lib\site-packages\ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """


CatalogNumber               Item Name     M.w.         Formula  \
0            A1042          Angiotensin II   1046.2    C50H71N13O12   
1            A1198           Levetiracetam   170.21       C8H14N2O2   
2            A1206              Daptomycin  1620.67   C72H101N17O26   
3            A1229            Lansoprazole   369.36   C16H14F3N3O2S   
4            A1267               Adapalene   412.52        C28H28O3   
...            ...                     ...      ...             ...   
1966         N2730     Chondroitin sulfate   489.45  (C14H21NO14S)n   
1967         N2758                 Aucubin   346.33        C15H22O9   
1968         N2771           Cepharanthine   606.71      C37H38N2O6   
1969        P10002     4-Aminobutyric acid   103.12         C4H9NO2   
1970        P10019  Prulifloxacin (Pruvel)   461.46    C21H20FN3O6S   

                                                 SMILES  \
0     CCC(C)C(C(=O)NC(CC1=CN=CN1)C(=O)N2CCCC2C(=O)NC...   
1                                  CCC(C(=O)N)N1CCCC1=O   
2     CCCCCCCCCC(=O)NC(CC1=CNC2=CC=CC=C21)C(=O)NC(CC...   
3       CC1=C(C=CN=C1CS(=O)C2=NC3=CC=CC=C3N2)OCC(F)(F)F   
4     COC1=C(C=C(C=C1)C2=CC3=C(C=C2)C=C(C=C3)C(=O)O)...   
...                                                 ...   
1966                                                 NA   
1967                                                 NA   
1968  CN1CCC2=CC3=C(C4=C2C1CC5=CC=C(C=C5)OC6=C(C=CC(...   
1969                                       C(CC(=O)O)CN   
1970  CC1N2C3=CC(=C(C=C3C(=O)C(=C2S1)C(=O)O)F)N4CCN(...   

                                                  URL  
0      http://www.apexbt.com/search.php?catalog=A1042  
1      http://www.apexbt.com/search.php?catalog=A1198  
2      http://www.apexbt.com/search.php?catalog=A1206  
3      http://www.apexbt.com/search.php?catalog=A1229  
4      http://www.apexbt.com/search.php?catalog=A1267  
...                                               ...  
1966   http://www.apexbt.com/search.php?catalog=N2730  
1967   http://www.apexbt.com/search.php?catalog=N2758  
1968   http://www.apexbt.com/search.php?catalog=N2771  
1969  http://www.apexbt.com/search.php?catalog=P10002  
1970  http://www.apexbt.com/search.php?catalog=P10019  

[1971 rows x 6 columns]

In [14]:
# Count the NA counts in smiles
na_smiles_count = sum(df_fda_selected.SMILES == 'NA')
na_smiles_count

22

## Retrieve unavailable SMILES manually

In [15]:
# Extract item name for the drugs with NA smiles
na_smiles_list = []
for i in range(len(df_fda_selected)):
    if df_fda_selected.SMILES[i] == 'NA':
        na_smiles_list.append(df_fda_selected['Item Name'][i])
na_smiles_list

['MK-2048',
 'Anisole Methoxybenzene',
 'Casanthranol',
 'Vancomycin hydrochloride',
 'Loperamide HCl',
 'Naloxone HCl',
 'Naltrexone HCl',
 'Colistin Sulfate',
 'Inulin',
 '4-Aminopyridine',
 'Pantoprazole',
 "2'-Deoxyguanosine",
 'Oxybenzone',
 'Zotarolimus(ABT-578)',
 'Brivaracetam',
 'Cetrorelix',
 'Carbetocin',
 'Thiostrepton',
 'Homoharringtonine',
 'Costundide',
 'Chondroitin sulfate',
 'Aucubin']

In [16]:
# Extract SMILES for the drugs with NA smiles manually
smiles_list = ['FC1=C(Cl)C=C(CN2C(C(C(O)=C3N4[C@@H](C)CN(CC)C3=O)=C4C(C(NC)=O)=N2)=O)C=C1',
               'COC1=CC=CC=C1',
               'C1=CC2=C(C(=C1)O)C(=O)C3=C(C2OC4C(C(C(C(O4)CO)O)O)O)C=C(C=C3O)CO',
               'CC1C(C(CC(O1)OC2C(C(C(OC2OC3=C4C=C5C=C3OC6=C(C=C(C=C6)C(C(C(=O)NC(C(=O)NC5C(=O)NC7C8=CC(=C(C=C8)O)C9=C(C=C(C=C9O)O)C(NC(=O)C(C(C1=CC(=C(O4)C=C1)Cl)O)NC7=O)C(=O)O)CC(=O)N)NC(=O)C(CC(C)C)NC)O)Cl)CO)O)O)(C)N)O.Cl',
               'CN(C)C(=O)C(CCN1CCC(CC1)(C2=CC=C(C=C2)Cl)O)(C3=CC=CC=C3)C4=CC=CC=C4',
               'C=CCN1CCC23C4C(=O)CCC2(C1CC5=C3C(=C(C=C5)O)O4)O',
               'C1CC1CN2CCC34C5C(=O)CCC3(C2CC6=C4C(=C(C=C6)O)O5)O',
               'CCC(C)CCCC(=O)NC(CCN)C(=O)NC(C(C)O)C(=O)NC(CCN)C(=O)NC1CCNC(=O)C(NC(=O)C(NC(=O)C(NC(=O)C(NC(=O)C(NC(=O)C(NC1=O)CCN)CC(C)C)CC(C)C)CCN)CCN)C(C)O',
               'C(C1C(C(C(C(O1)OC2(C(C(C(O2)CO)O)O)COC3(C(C(C(O3)CO)O)O)COC4(C(C(C(O4)CO)O)O)COC5(C(C(C(O5)CO)O)O)COC6(C(C(C(O6)CO)O)O)COC7(C(C(C(O7)CO)O)O)COC8(C(C(C(O8)CO)O)O)COC9(C(C(C(O9)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)COC1(C(C(C(O1)CO)O)O)CO)O)O)O)O',
               'C1=CN=CC=C1N',
               'COC1=C(C(=NC=C1)CS(=O)C2=NC3=C(N2)C=C(C=C3)OC(F)F)OC',
               'C1C(C(OC1N2C=NC3=C2N=C(NC3=O)N)CO)O',
               'COC1=CC(=C(C=C1)C(=O)C2=CC=CC=C2)O',
               'CC1CCC2CC(C(=CC=CC=CC(CC(C(=O)C(C(C(=CC(C(=O)CC(OC(=O)C3CCCCN3C(=O)C(=O)C1(O2)O)C(C)CC4CCC(C(C4)OC)N5C=NN=N5)C)C)O)OC)C)C)C)OC',
               'CCCC1CC(=O)N(C1)C(CC)C(=O)N',
               'CC(C)CC(C(=O)NC(CCCN=C(N)N)C(=O)N1CCCC1C(=O)NC(C)C(=O)N)NC(=O)C(CCCNC(=O)N)NC(=O)C(CC2=CC=C(C=C2)O)NC(=O)C(CO)NC(=O)C(CC3=CN=CC=C3)NC(=O)C(CC4=CC=C(C=C4)Cl)NC(=O)C(CC5=CC6=CC=CC=C6C=C5)NC(=O)C',
               'CCC(C)C1C(=O)NC(C(=O)NC(C(=O)NC(CSCCCC(=O)NC(C(=O)N1)CC2=CC=C(C=C2)OC)C(=O)N3CCCC3C(=O)NC(CC(C)C)C(=O)NCC(=O)N)CC(=O)N)CCC(=O)N',
               'CCC(C)C1C(=O)NC(C(=O)NC(=C)C(=O)NC(C(=O)NC23CCC(=NC2C4=CSC(=N4)C(C(OC(=O)C5=NC6=C(C=CC(C6O)N1)C(=C5)C(C)O)C)NC(=O)C7=CSC(=N7)C(NC(=O)C8CSC(=N8)C(=CC)NC(=O)C(NC(=O)C9=CSC3=N9)C(C)O)C(C)(C(C)O)O)C1=NC(=CS1)C(=O)NC(=C)C(=O)NC(=C)C(=O)N)C)C',
               'CC(C)(CCCC(CC(=O)OC)(C(=O)OC1C2C3=CC4=C(C=C3CCN5C2(CCC5)C=C1OC)OCO4)O)O',
               'CC1=CCCC(=CC2C(CC1)C(=C)C(=O)O2)C',
               'CC(=O)NC1C(C(C(OC1O)CO)O)OC2C(C(C=C(O2)C(=O)O)O)O',
               'C1=COC(C2C1C(C=C2CO)O)OC3C(C(C(C(O3)CO)O)O)O'
            ]
len(smiles_list)

22

In [17]:
# Add the smiles to the df_fda_selected dataframe
df_na_smiles = pd.DataFrame({'Item Name': na_smiles_list, 'SMILES': smiles_list})
df_na_smiles

Item Name  \
0                    MK-2048   
1     Anisole Methoxybenzene   
2               Casanthranol   
3   Vancomycin hydrochloride   
4             Loperamide HCl   
5               Naloxone HCl   
6             Naltrexone HCl   
7           Colistin Sulfate   
8                     Inulin   
9            4-Aminopyridine   
10              Pantoprazole   
11         2'-Deoxyguanosine   
12                Oxybenzone   
13      Zotarolimus(ABT-578)   
14              Brivaracetam   
15                Cetrorelix   
16                Carbetocin   
17              Thiostrepton   
18         Homoharringtonine   
19                Costundide   
20       Chondroitin sulfate   
21                   Aucubin   

                                               SMILES  
0   FC1=C(Cl)C=C(CN2C(C(C(O)=C3N4[C@@H](C)CN(CC)C3...  
1                                       COC1=CC=CC=C1  
2   C1=CC2=C(C(=C1)O)C(=O)C3=C(C2OC4C(C(C(C(O4)CO)...  
3   CC1C(C(CC(O1)OC2C(C(C(OC2OC3=C4C=C5C=C3OC6=C(C...  
4   CN(C)C(=O)C(CCN1CCC(CC1)(C2=CC=C(C=C2)Cl)O)(C3...  
5     C=CCN1CCC23C4C(=O)CCC2(C1CC5=C3C(=C(C=C5)O)O4)O  
6   C1CC1CN2CCC34C5C(=O)CCC3(C2CC6=C4C(=C(C=C6)O)O5)O  
7   CCC(C)CCCC(=O)NC(CCN)C(=O)NC(C(C)O)C(=O)NC(CCN...  
8   C(C1C(C(C(C(O1)OC2(C(C(C(O2)CO)O)O)COC3(C(C(C(...  
9                                        C1=CN=CC=C1N  
10  COC1=C(C(=NC=C1)CS(=O)C2=NC3=C(N2)C=C(C=C3)OC(...  
11                C1C(C(OC1N2C=NC3=C2N=C(NC3=O)N)CO)O  
12                 COC1=CC(=C(C=C1)C(=O)C2=CC=CC=C2)O  
13  CC1CCC2CC(C(=CC=CC=CC(CC(C(=O)C(C(C(=CC(C(=O)C...  
14                        CCCC1CC(=O)N(C1)C(CC)C(=O)N  
15  CC(C)CC(C(=O)NC(CCCN=C(N)N)C(=O)N1CCCC1C(=O)NC...  
16  CCC(C)C1C(=O)NC(C(=O)NC(C(=O)NC(CSCCCC(=O)NC(C...  
17  CCC(C)C1C(=O)NC(C(=O)NC(=C)C(=O)NC(C(=O)NC23CC...  
18  CC(C)(CCCC(CC(=O)OC)(C(=O)OC1C2C3=CC4=C(C=C3CC...  
19                  CC1=CCCC(=CC2C(CC1)C(=C)C(=O)O2)C  
20  CC(=O)NC1C(C(C(OC1O)CO)O)OC2C(C(C=C(O2)C(=O)O)O)O  
21       C1=COC(C2C1C(C=C2CO)O)OC3C(C(C(C(O3)CO)O)O)O

In [20]:
# Merge the df_train_selection and df_fda_selected
for i in range(len(df_na_smiles)):
    for j in range(len(df_fda_selected)):
        if df_na_smiles['Item Name'][i] == df_fda_selected['Item Name'][j]:
            df_fda_selected.SMILES[j] = df_na_smiles.SMILES[i]

c:\Anaconda3\envs\dleps\lib\site-packages\ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """


False    1971
Name: SMILES, dtype: int64

In [22]:
sum(df_fda_selected.SMILES.isna()==True)

0

**Finally, there are no empty SMILES in the dataset**

In [24]:
# Save the merged dataframe as csv file
df_merged = pd.merge(df_train_selection, df_fda_selected, left_on='DRUG NAME', right_on='Item Name')
df_merged

DRUG NAME  MDA7+D1+1  MDA7+D1+2  MDA7+D1+3  MDA7+D2+1  \
0             Angiotensin II   0.467198   0.492122   0.500198   0.651233   
1              Levetiracetam   0.424844   0.503705   0.433523   0.517083   
2                 Daptomycin   0.462072   0.509482   0.424083   0.451883   
3               Lansoprazole   0.464806   0.494925   0.502533   0.491361   
4                  Adapalene   0.407362   0.425062   0.490854   0.489590   
...                      ...        ...        ...        ...        ...   
1960     Chondroitin sulfate   0.750820   0.741770   0.826674   0.835227   
1961                 Aucubin   0.714064   0.849337   0.563837   0.869225   
1962           Cepharanthine   0.592418   0.708048   0.883598   0.811905   
1963     4-Aminobutyric acid   0.705559   0.941525   0.717699   0.887134   
1964  Prulifloxacin (Pruvel)   0.844985   0.773646   0.742799   0.802596   

      MDA7+D2+2  MDA7+D2+3 CatalogNumber               Item Name     M.w.  \
0      0.762780   0.484070         A1042          Angiotensin II   1046.2   
1      0.498133   0.485443         A1198           Levetiracetam   170.21   
2      0.495565   0.472871         A1206              Daptomycin  1620.67   
3      0.479975   0.443513         A1229            Lansoprazole   369.36   
4      0.513733   0.395002         A1267               Adapalene   412.52   
...         ...        ...           ...                     ...      ...   
1960   0.807444   0.802552         N2730     Chondroitin sulfate   489.45   
1961   0.700734   0.823960         N2758                 Aucubin   346.33   
1962   0.849278   0.827013         N2771           Cepharanthine   606.71   
1963   0.716770   0.617353        P10002     4-Aminobutyric acid   103.12   
1964   0.787334   0.757312        P10019  Prulifloxacin (Pruvel)   461.46   

             Formula                                             SMILES  \
0       C50H71N13O12  CCC(C)C(C(=O)NC(CC1=CN=CN1)C(=O)N2CCCC2C(=O)NC...   
1          C8H14N2O2                               CCC(C(=O)N)N1CCCC1=O   
2      C72H101N17O26  CCCCCCCCCC(=O)NC(CC1=CNC2=CC=CC=C21)C(=O)NC(CC...   
3      C16H14F3N3O2S    CC1=C(C=CN=C1CS(=O)C2=NC3=CC=CC=C3N2)OCC(F)(F)F   
4           C28H28O3  COC1=C(C=C(C=C1)C2=CC3=C(C=C2)C=C(C=C3)C(=O)O)...   
...              ...                                                ...   
1960  (C14H21NO14S)n  CC(=O)NC1C(C(C(OC1O)CO)O)OC2C(C(C=C(O2)C(=O)O)O)O   
1961        C15H22O9       C1=COC(C2C1C(C=C2CO)O)OC3C(C(C(C(O3)CO)O)O)O   
1962      C37H38N2O6  CN1CCC2=CC3=C(C4=C2C1CC5=CC=C(C=C5)OC6=C(C=CC(...   
1963         C4H9NO2                                       C(CC(=O)O)CN   
1964    C21H20FN3O6S  CC1N2C3=CC(=C(C=C3C(=O)C(=C2S1)C(=O)O)F)N4CCN(...   

                                                  URL  
0      http://www.apexbt.com/search.php?catalog=A1042  
1      http://www.apexbt.com/search.php?catalog=A1198  
2      http://www.apexbt.com/search.php?catalog=A1206  
3      http://www.apexbt.com/search.php?catalog=A1229  
4      http://www.apexbt.com/search.php?catalog=A1267  
...                                               ...  
1960   http://www.apexbt.com/search.php?catalog=N2730  
1961   http://www.apexbt.com/search.php?catalog=N2758  
1962   http://www.apexbt.com/search.php?catalog=N2771  
1963  http://www.apexbt.com/search.php?catalog=P10002  
1964  http://www.apexbt.com/search.php?catalog=P10019  

[1965 rows x 13 columns]

In [25]:
# Save the merged dataframe as csv file
df_merged.to_csv('../../data/training_dataset_clean.csv', index=False)

## **Finally, this dataset is ready for further preprocessing - for both Encoder model, and FC model!**